abstract-level.js

```diff
@@ -34,14 +35,18 @@ function AbstractLevel (manifest, options) {
   options = getOptions(options)
   EventEmitter.call(this)
 
-  const { keyEncoding, valueEncoding, passive, ...forward } = options
+  const { keyEncoding, valueEncoding, passive, prime, ...forward } = options
 
   this[kResources] = new Set()
   this[kOperations] = []
   this[kDeferOpen] = true
   this[kOptions] = forward
   this[kStatus] = 'opening'
 
+  // Undocumented and experimental option.
+  // TODO: benchmark
+  this[kPrime] = prime === true
```

Comment:
Remove this. The basic idea of this new "prime" feature was that if you have an `abstract-level` layer wrapping another (like how a sublevel wraps the parent db), let's call them a shell and a nut, and you do a `batch(operations)` for example. Then only the shell would have to iterate the `operations` array, encode keys & values, normalize options, etc. The nut could just take arguments as-is, e.g. because they're already encoded.

I did a quick benchmark and it does result in a 10-20% speedup, but we're talking nanoseconds. With _prime_ enabled, a batch of 1000 puts took 1ms (on `memory-level`, the replacement for `memdown`). With _prime_ disabled, it took 0.9ms. That said, the benefit on real-world apps would be bigger than on a synthetic benchmark, due to less GC pressure (not having to clone operations). Maybe I'll revisit it some day, but for now it's not worth the added complexity (and potential side effects).

Reply:
Not to mention that merging `encoding-down` into `abstract-level` already removed two iterations of the batch operations array, where each was deeply cloning that array, and merging `subleveldown` into `abstract-level` removes another iteration (which didn't clone). So previously (not including C++) every batch array was iterated 5 times, now 2 times.